src/model/SubSurface.cpp

```diff
@@ -603,6 +615,31 @@ namespace model {
           }
         }
 
+        if (!allowDaylightingDeviceTubularDome()) {
+          boost::optional<DaylightingDeviceTubular> tubular = this->daylightingDeviceTubular();
+          if (tubular) {
+            if (tubular->subSurfaceDome().handle() == this->handle()) {
+              tubular->remove();
+            }
+          }
+        }
```

Comment:
I know you're just following the existing pattern, but I think we should Warn in all of these cases no? Might be confusing for a user to see stuff disappear from the model...